python/cudf/cudf/core/dataframe.py

```diff
@@ -600,14 +600,16 @@ def __setitem__(self, arg, value):
                         mask=None,
                     )
                 else:
-                    if not is_scalar(value):
-                        value = column.as_column(value)
                     for col in arg:
                         # we will raise a key error if col not in dataframe
                         # this behavior will make it
                         # consistent to pandas >0.21.0
                         if not is_scalar(value):
-                            self._data[col] = value
+                            self._data[col] = column.as_column(value)
+                        elif col not in self._data.keys():
+                            self._data[col] = column.as_column(
```

Comment:
Let's use `column.as_column(value, length=len(self))` instead of calling `utils.scalar_broadcast_to(...)`.

Reply:
`as_column` is also calling broadcast, but that would make future proof